src/libtriton/engines/solver/bitwuzla/bitwuzlaSolver.cpp

```diff
@@ -0,0 +1,243 @@
+//! \file
+/*
+**  Copyright (C) - Triton
+**
+**  This program is under the terms of the Apache License 2.0.
+*/
+
+#include <fstream>
+#include <regex>
+#include <string>
+
+#include <triton/astContext.hpp>
+#include <triton/bitwuzlaSolver.hpp>
+#include <triton/exceptions.hpp>
+#include <triton/solverModel.hpp>
+#include <triton/symbolicExpression.hpp>
+#include <triton/symbolicVariable.hpp>
+#include <triton/tritonToBitwuzlaAst.hpp>
+#include <triton/tritonTypes.hpp>
+
+
+namespace triton {
+  namespace engines {
+    namespace solver {
+
+      int32_t BitwuzlaSolver::terminate_fun(void* state) {
+        auto p = reinterpret_cast<SolverParams*>(state);
+
+        // Check time only once in every 1000 calls.
+        if (++p->call_cnt < 1000) {
+          return 0;
+        }
+        p->call_cnt = 0;
+
+        // Count delta.
+        auto delta = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now() - p->start).count();
+
+        // Check timeout expired.
+        if (p->timeout && delta > p->timeout) {
+          p->code = 1;
+          return 1;
+        }
+
+        // Check memory limit.
+        delta /= 1000;
+        if (p->memory_limit && delta > p->last_mem_check && delta % p->delay == 0) {
```

Comment:
delta > p->last_mem_check is signed/unsigned comparison? maybe better check delta != last_mem_check

Reply:
it is signed comparison to make first check more convenient